### ISSUE TITLE
[Tasks] Pre-register Custom Tasks

### ISSUE THREAD

**Issue by andres-montanez** [andres-montanez/Magallanes#316]
Allow to pre-register custom tasks on the config file so if the task is used many times on the same project then it can be invoked by its "name" and not by it's classname

Example:
```yaml
magephp:
    environments:
        production:
            post-deploy:
                - do-magic
        staging:
            post-deploy:
                - do-magic
        testing:
            post-deploy:
                - do-magic
        demo:
            post-deploy:
                - do-magic
    custom_tasks:
        - AppBundle\Library\Deployment\DoMagicTask
```

```php
namespace AppBundle\Library\Deployment;

use Mage\Task\AbstractTask;

class DoMagicTask extends AbstractTask
{
    public function getName()
    {
        return 'do-magic';
    }

    public function getDescription()
    {
        return '[Custom] Doing some magic';
    }

    public function execute()
    {
        /* some awesome magic happening here */
    }
}
```

At init time all registered custom tasks will be instantiated and added just like the built in tasks.
This should make the config file more readable, and also if the class needs to be moved then you just only have to change it in one place.

**Comment by dn5** [comment]
Looking forward to see this. Did you think about the way you are planning to implement this?

**Comment by andres-montanez** [comment]
Hi @dn5 like in the issue description, and load the tasks at runtime after the built in tasks.

**Comment by andres-montanez** [comment]
Implemented in version 3.2
